perf/reader.go

```diff
@@ -45,6 +45,8 @@ type Record struct {
 	// garbage from the ring depending on the input sample's length.
 	RawSample []byte
 
+	RawEvent []byte
```

Comment:
Could we just reuse RawSample instead of introducing a new slice?

If this is required it should be documented.

Reply:
so RawSample actually contains specific data for sample event PERF_COUNT_SW_BPF_OUTPUT:

```
perfEventHeader {
        Type uint32
        Misc uint16
        Size uint16
}
sample (perfEventHeader.Type == 9) {
        Size uint32
        RawSample []byte
}
```

for events other than sample above (like MMAP/COMM/FORK or SAMPLES for events other than bpf-output)
we need to return just the full byte array:

```
perfEventHeader {
        Type uint32
        Misc uint16
        Size uint16
}
event (perfEventHeader.Type == *) {
        RawEvent []byte
}
```

I can return it in the rec.RawSample, but I'd still prefer to add rec.RawEvent
also we need separate read function to do that